mesh/mesh.hpp

```diff
@@ -29,6 +29,7 @@
 #include <iostream>
 #include <array>
 #include <map>
+#include <memory>
```

Comment:
I was able to build without this include. It looks like it is included for `unique_ptr`, but it seems to be included from somewhere else.

Reply:
One sequence of includes that brings in `<memory>` in `mesh.hpp` is:

    mesh.hpp -> eltrans.hpp -> linalg.hpp -> solvers.hpp -> memory

so, technically, we can remove the direct inclusion of `<memory>`. However, it maybe better the keep it for clarity. What do you prefer? I don't have a strong preference.

Reply:
I also don't have a strong preference.

Reply:
I would keep it, in the spirit of https://google.github.io/styleguide/cppguide.html#Include_What_You_Use.

Reply:
Ok, I'll keep it.